Managers/PrimitiveMan.h

```diff
@@ -0,0 +1,546 @@
+#ifndef _RTEPRIMITIVEMAN_
+#define _RTEPRIMITIVEMAN_
+
+#include "Singleton.h"
+#include "Vector.h"
+
+#define g_PrimitiveMan PrimitiveMan::Instance()
+
+struct BITMAP;
+
+namespace RTE {
+
+	class Entity;
+
+	/// <summary>
+	/// Singleton manager responsible for all primitive drawing.
+	/// </summary>
+	class PrimitiveMan : public Singleton<PrimitiveMan> {
+
+	public:
+
+#pragma region Primitive Objects
+#pragma region Graphical Primitive
+		/// <summary>
+		/// Class used to schedule drawing of graphical primitives created from Lua.
+		/// All coordinates passed to GraphicalPrimitive objects are Scene coordinates.
+		/// </summary>
+		class GraphicalPrimitive {
+
+		public:
+
+			Vector m_StartPos; //!< Start position of the primitive.
+			Vector m_EndPos; //!< End position of the primitive.
+			unsigned short m_Color; //!< Color to draw this primitive with.		
+			short m_Player; //!< Player screen to draw primitive on.
+
+			/// <summary>
+			/// Constructor method for GraphicalPrimitive objects.
+			/// </summary>
+			GraphicalPrimitive() { m_Color = 0; m_Player = -1; }
+
+			/// <summary>
+			/// Translates coordinates from scene to this bitmap offset.
+			/// </summary>
+			/// <param name="targetPos">Bitmap to draw on.</param>
+			/// <param name="scenePos">Bitmap offset (scene coordinates).</param>
+			/// <returns>Vector with translated coordinates.</returns>
+			Vector TranslateCoordinates(Vector targetPos, Vector scenePos) { return scenePos - targetPos; }
```

Comment:
I think this TranslateCoordinates is totally unused (do shift+F12 to check for references), which would make sense cause it doesn't account for SceneWrapping properly.